src/Azure.Functions.Cli/Helpers/VersionHelper.cs

```diff
@@ -0,0 +1,141 @@
+﻿using Azure.Functions.Cli.Common;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json.Nodes;
+using System.Threading.Tasks;
+
+namespace Azure.Functions.Cli.Helpers
+{
+    internal class VersionHelper
+    {
+
+        // Check that current core tools is the latest version. 
+        // To ensure that it doesn't block other tasks. The HTTP Request timeout is only 1 second. 
+        // We simply ingnore the exception if for any reason the check fails. 
+        public static async Task<string> CheckLatestVersionAsync()
+        {
+            try
+            {
+                var client = new System.Net.Http.HttpClient
+                {
+                    Timeout = TimeSpan.FromSeconds(1)
+                };
+                var response = await client.GetAsync(Constants.CoreToolsVersionsFeedUrl);
+                var content = await response.Content.ReadAsStringAsync();
+                var data = JsonConvert.DeserializeObject<CliFeed>(content);
+                IEnumerable releases = ((IEnumerable) data.Releases);
+                var releaseList = new List<ReleaseSummary>();
+                foreach (var item in releases)
+                {
+                    var jProperty = (Newtonsoft.Json.Linq.JProperty)item;
+                    var releaseDetail = JsonConvert.DeserializeObject<ReleaseDetail>(jProperty.Value.ToString());
+                    releaseList.Add(new ReleaseSummary() { Release = jProperty.Name, ReleaseDetail = releaseDetail.ReleaseList.FirstOrDefault() });
+                }
+
+                var currentCoreToolsRelease = releaseList.FirstOrDefault(x => x.CoreToolsReleaseNumber == Constants.CliDetailedVersion[..Constants.CliDetailedVersion.IndexOf(" ")]);
```

Comment:
Could we use `FileVersion.GetVersionInfo` to create the value of Constants.CliDetailedVersion? In that case, I don't think taking a substring would be necessary.